source/secure_tunneling.c

```diff
@@ -189,12 +193,39 @@ static int s_secure_tunneling_send_data(struct aws_secure_tunnel *secure_tunnel,
     x->secure_tunnel = secure_tunnel;
     x->buf = &buffer;
 
+    if (buffer.len > MAX_ST_PAYLOAD) {
+        return AWS_OP_ERR;
+    }
     struct aws_websocket_send_frame_options frame_options;
     s_init_websocket_send_frame_options(&frame_options, x);
     aws_websocket_send_frame(secure_tunnel->websocket, &frame_options);
     return AWS_OP_SUCCESS;
 }
 
+static int s_secure_tunneling_send_data(struct aws_secure_tunnel *secure_tunnel, const struct aws_byte_buf *data) {
+    if (secure_tunnel == NULL || secure_tunnel->stream_id == INVALID_STREAM_ID) {
+        return AWS_OP_ERR;
+    }
+    return s_secure_tunneling_send(secure_tunnel, data, DATA);
+}
+
+static int s_secure_tunneling_send_stream_start(struct aws_secure_tunnel *secure_tunnel) {
+    if (secure_tunnel == NULL || secure_tunnel->stream_id == INVALID_STREAM_ID ||
+        secure_tunnel->config.local_proxy_mode == AWS_SECURE_TUNNELING_DESTINATION_MODE) {
+        return AWS_OP_ERR;
+    }
+    struct aws_byte_buf buffer = aws_byte_buf_from_c_str("");
+    return s_secure_tunneling_send(secure_tunnel, &buffer, STREAM_START);
+}
```

Comment:
Couple things:

1. The condition to test `secure_tunnel->stream_id == INVALID_STREAM_ID` should be removed
2. I think you need to update/increment stream id
3. Consider passing NULL as the byte buffer when calling `s_secure_tunneling_send`